src/tools/perf/perftest.c

```diff
@@ -460,6 +466,31 @@ static ucs_status_t parse_ucp_datatype_params(const char *optarg,
     return UCS_OK;
 }
 
+static ucs_status_t parse_mem_type_params(const char *optarg,
+                                          ucs_memory_type_t *mem_type)
+{
+    if (!strncmp(optarg, "cuda-managed", 12) &&
```

Comment:
maybe add a wrapper for `strncmp()` that calculates a length of passed memory type and compare with `optarg`?